dams-client/src/api/arbitrary_secrets/generate.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{DamsClient, DamsClientError};
+use dams::{
+    channel::ClientChannel,
+    crypto::{KeyId, Secret, StorageKey},
+    types::generate::{client, server},
+    user::UserId,
+};
+use rand::rngs::StdRng;
+
+impl DamsClient {
+    pub(crate) async fn handle_generate(
+        &self,
+        channel: &mut ClientChannel,
+    ) -> Result<(KeyId, Secret), DamsClientError> {
+        // Retrieve the storage key
+        let storage_key = self.retrieve_storage_key().await?;
+
+        // Generate step: get new KeyId from server
+        let key_id = generate(channel, self.user_id()).await?;
+        // Store step: encrypt secret and send to server to store
+        let secret = {
+            let rng = self.rng();
+            let mut rng = rng.lock().await;
```

Comment:
```suggestion
            let mut rng = self.rng.lock().await;
```
If we make the rng public we can do this.